cmd/demo-client/main.go

```diff
@@ -159,3 +206,53 @@ func Watch(c *cli.Context, client client.Client) error {
 	}
 	return nil
 }
+
+func newPrometheusBridge(address string, gateway string, pushIntervalSec int64) prometheus.Registerer {
+	b := &prometheusBridge{
+		address:         address,
+		pushIntervalSec: pushIntervalSec,
+		registry:        prometheus.NewRegistry(),
+	}
+	if gateway != "" {
+		b.pusher = push.New(gateway, "drand_client_observations_push")
+		go b.pushLoop()
+	}
+	if address != "" {
+		http.Handle("/metrics", promhttp.HandlerFor(b.registry, promhttp.HandlerOpts{
+			Timeout: 10 * time.Second,
+		}))
+		go log.Fatal(http.ListenAndServe(address, nil))
+	}
+	return b
+}
+
+type prometheusBridge struct {
+	address         string
+	pushIntervalSec int64
+	registry        *prometheus.Registry
+	pusher          *push.Pusher
+}
+
+func (b *prometheusBridge) pushLoop() {
+	for {
```

Comment:
have a way to cancel / life-time this?

Reply:
Why? There is no use case for this.

Reply:
so that as we make tests for the client CLI they can cancel the context they've provided and have everything associated with it shut down

Reply:
I think this is out of scope for this PR, because making the client be able to Shutdown requires a lot more than just fixing this pushLoop. None of the client components have shutdown logic yet. Fixing this should belong in a separate PR, I think.